File: src/net/slreynolds/ds/IntScalaHashMapTiming.java
package net.slreynolds.ds;


import scala.Tuple2;
import scala.collection.immutable.HashMap;
import scala.collection.immutable.IntMap;

public class IntScalaHashMapTiming extends AbstractTiming<HashMap> {


    @Override
    protected int doWork(HashMap map) {
        
        final Tuple2[] vals = IntMapTimingParameters.valuesToInsert;
        final int n = IntMapTimingParameters.numToInsert;
        for (int i = 0; i < n; i++) {
            map = map.$plus(vals[i]);
        }
        return map.size();
    }
    
    @Override
    protected HashMap setUp() {
        HashMap baseHMap = new HashMap();
        for (int i = 0; i < IntMapTimingParameters.initialSize; i++) {
            baseHMap = baseHMap.$plus(new Tuple2(i,new SomeValue(i)));
        }
        return baseHMap;
    }
    
    public static void main(String[] args) { 
        new IntScalaHashMapTiming().run("Int Scala HashMap");
    }


}

File: src/net/slreynolds/ds/IntScalaHashMapTiming.scala
package net.slreynolds.ds


import scala.Tuple2;
import scala.collection.immutable.HashMap;

object IntScalaHashMapTiming extends AbstractTiming[HashMap[Int,SomeValue], HashMap[Int,SomeValue]] {


	
    @Override
    def doWork(map: HashMap[Int,SomeValue]  ): Result[HashMap[Int,SomeValue]] = {
        var lmap : HashMap[Int,SomeValue] = map
        val vals = IntMapSource.valuesToInsert
        val n = IntMapTimingParameters.numToInsert
        for (i <- 0 to n-1)  {
            lmap += (vals(i))
        }
        new Result(lmap,lmap.size)
    }
    
    @Override
    def setUp() : HashMap[Int,SomeValue] = {
        var baseHMap : HashMap[Int,SomeValue] = new HashMap();
        for (i <- 0 to IntMapTimingParameters.initialSize-1)  {
             baseHMap += (new Tuple2(i,new SomeValue(i)))
        }
        baseHMap
    }
    
    def main(args: Array[String]): Unit = {
        run("Int Scala immutable HashMap")
    }

}